python/ray/local_mode_manager.py

```diff
@@ -49,23 +61,37 @@ def execute(self, function, function_name, args, kwargs, num_return_vals):
         Returns:
             LocalModeObjectIDs corresponding to the function return values.
         """
-        object_ids = [
+        return_ids = [
             LocalModeObjectID.from_random() for _ in range(num_return_vals)
         ]
+        new_args = []
+        for i, arg in enumerate(args):
+            if isinstance(arg, ObjectID):
+                new_args.append(ray.get(arg))
+            else:
+                new_args.append(copy.deepcopy(arg))
```

Comment:
does it make sense to push this through plasma serialization? would that make this more realistic?

Reply:
yes but this is not totally trivial so let's leave it to another patch